docs: updated README.md to include link to repo, files list and description

cisba opened the pull request:
This fixes #8160 (partially), fixes #8161, fixes #11971

cisba committed:
[Docs]: updated README.md to be included into unix tar archives, containing files list and description.

    This fixes #8160, fixes #8161, fixes #11971

cisba committed:
[Docs]: updated README.md to be included into unix tar archives, containing link to repo, files list and description.

    This fixes #8160, fixes #8161, fixes #11971

laanwj commented:
I'm not sure this fixes #8161? Does this add a link to the source repository so that people can find the source code easier?

cisba commented:
Added the link to repo.

fanquake commented on doc/README.md:
```diff
@@ -81,4 +98,4 @@ License
 Distributed under the [MIT software license](/COPYING).
 This product includes software developed by the OpenSSL Project for use in the [OpenSSL Toolkit](https://www.openssl.org/). This product includes
-cryptographic software written by Eric Young ([eay@cryptsoft.com](mailto:eay@cryptsoft.com)), and UPnP software written by Thomas Bernard.
+cryptographic software written by Eric Young ([eay@cryptsoft.com](mailto:eay@cryptsoft.com)), and UPnP software written by Thomas Bernard.
```
What is changing here?

AkioNak replied:
It seems that the last '\n' has been deleted.

cisba replied:
Yes, not intentional but also not relevant (IMHO)...

cisba replied:
@fanquake I restored the last '\n' and now it seem like I added a blank line... LOL

cisba replied:
fixed

cisba committed:
docs: typos fixed

PR Title:
    docs: updated README.md to include link to repo, files list and
    description

PR Description:
    This fixes #8160 (partially), fixes #8161, see also PR #11971 and #14022

cisba committed:
docs: typos

hebasto commented:
> For the sake of style consistency all executables names should be either quoted within the backticks or not quoted at all.

I meant `bitcoin-qt.exe` in the Running Windows section as well.

cisba committed:
docs: bitcoin-qt.exe quoted as the others filenames

cisba committed:
docs: typo

cisba committed:
docs: typo

cisba commented on doc/README.md:
```diff
@@ -22,7 +22,7 @@ Unpack the files into a directory and run:
 
 ### Windows
 
-Unpack the files into a directory, and then run 
```
trailing whitespace generated error in `test/lint/lint-whitespace.sh`
https://travis-ci.org/bitcoin/bitcoin/jobs/425139871

cisba commented:
@laanwj I think all points now are checked. Anyway, I made a new PR #14167 with the aim to improve even more the documentation in tar packages.

merland commented:
In [PR 14755](https://github.com/bitcoin/bitcoin/pull/14755), I suggest to remove all instructions about running and instead link to the official docs, to avoid duplication. Currently in conflict with this PR. Review/comment requested.  
(Also related: #14639)

fanquake commented:
Closing this, given that #14167 includes these changes.
